### ISSUE TITLE
Suggested improvements

### ISSUE THREAD

**Issue by raquelmsmith** [PostHog/nps-survey-app#1]
### Remove the black border

The black border isn't necessary and looks clumsy.

### Don't use color for the circles

The colored circles also look a bit aggressive. I'd use the same color for each background - maybe a light gray. If you really want the color then just have it show up on hover? @corywatilo can probably advise on design a bit more here too. 

### Use a for loop to generate the buttons

Each of these buttons are exactly the same except for the number inside them. 

https://github.com/PostHog/nps-survey-app/blob/89435cf8da760414a60ae1f7586071801a5104a9/site.ts#L173-L185

Conveniently for loops are really good at handling incrementally increasing numbers! I'd recommend doing something like the following:

```
for (i = 0, i++, i <= 10) {
    // your button code here
}
```

### Using with feature flags

Using a feature flag to show this is a good idea! To do that you'd just check for the flag before doing any of the injection. Eg here: https://github.com/PostHog/nps-survey-app/blob/89435cf8da760414a60ae1f7586071801a5104a9/site.ts#L104-L107

Right as the function opens, you'd do:

```
if (window.posthog && window.posthog.getFeatureFlag('posthog-nps-survey-app') === true) {
    // app code
)
```

It seems like not everything is running inside this inject function so you may need to do some rearranging if this doesn't work at the get-go.

You'd have to instruct people to create a feature flag using that name and set it to true for people who they want to see it.

You _also_ need to make sure people are only asked for their response once. Luke's interview scheduler should have an example of how to do this - he sets a property on the person (I think) with the flag name and some value. Then in the `posthog-nps-survey-app` flag you only show it to people who don't have that property set. 

### Other things

If you're hiding the container you shouldn't need to hide the buttons themselves as you do here:

https://github.com/PostHog/nps-survey-app/blob/89435cf8da760414a60ae1f7586071801a5104a9/site.ts#L167-L169

Also good to remove any unnecessary logging: https://github.com/PostHog/nps-survey-app/blob/89435cf8da760414a60ae1f7586071801a5104a9/site.ts#L134

If we want to show the PostHog name longer, we could show it initially below the rating buttons. Not sure what it looks like (haven't run this locally) but we'd want to make it minimal (light gray). 

https://github.com/PostHog/nps-survey-app/blob/89435cf8da760414a60ae1f7586071801a5104a9/site.ts#L145-L149

**Comment by joethreepwood** [comment]
- [x] Change the border
- [x] Change the circle colour
- [ ] ~Use a loop~
- [ ] Add flags
- [x] Refactor the container
- [x] Remove logging
- [ ] ~Show the brand longer~

Raquel, this is amazing feedback and I _massively_ appreciate it, thank you!

I've tried to add a loop previously on Eli's advice, but couldn't get it working so will skip that for now. My understanding is that it will make the code neater and faster, but on something this size won't create a real user advantage. So, good to know it's there but not needed rn. 

I'll also leave the PostHog name as-is, since I'm only really adding that in at Tim's guidance - I'm sure users will actually prefer this to be unbranded, so I want to avoid being heavy-handed. 

The FF stuff is proving tricky. If I understand correctly, this should work: 

```
export function inject({ config, posthog }) 

if (window.posthog && window.posthog.getFeatureFlag('posthog-nps-survey-app') === true)

{
  const shadow = createShadow(style);
// rest of app code
```
And I can toggle `true` to `false` to make it appear locally without setting the flag. However, this doesn't seem to work on local, regardless of the toggle. 

Likewise, I'd already looked at the interview app to try and puzzle out the once-only flag and got a bit lost with it, tbh!

**Comment by joethreepwood** [comment]
New UI 

https://github.com/PostHog/nps-survey-app/assets/84011561/5d513888-22c8-453c-b274-84b61eff9372

**Comment by raquelmsmith** [comment]
> I've tried to add a loop previously on Eli's advice, but couldn't get it working so will skip that for now. My understanding is that it will make the code neater and faster, but on something this size won't create a real user advantage. So, good to know it's there but not needed rn.

True, not a real user advantage, but good for learning :) If you want to put up a PR with your attempt I can see if I can spot the issue.

> And I can toggle true to false to make it appear locally without setting the flag. However, this doesn't seem to work on local, regardless of the toggle.

So wait, can you make it work locally or no 😅 How are you trying to see if it works?

> Change the border

I don't think it needs a border at all if it's inside a white container like your video shows? Or is that marketing-fu and not how it actually appears on a site? Can you include an actual screenshot if it's different on a site from your video?

**Comment by joethreepwood** [comment]
> So wait, can you make it work locally or no 😅 How are you trying to see if it works?

I've added in the FF code you provided and tried toggling it to true/false and a few other variations - but in each case, it just kills the app when running locally. Some error makes it think there's no site.ts to pull from. 

> I don't think it needs a border at all if it's inside a white container like your video shows? Or is that marketing-fu and not how it actually appears on a site? Can you include an actual screenshot if it's different on a site from your video?

No, that's how it appears. I prefer the border at the moment, as I'm aware it may be show on sites with different designs where the extra distinction is useful.

**Comment by raquelmsmith** [comment]
Can you share the error you get when trying to add the FF code?

**Comment by corywatilo** [comment]
+1 on removing border (or at least drastically reducing its boldness. Eg: `rgba(0,0,0,.25)` and maybe even a slight border radius (`border-radius: 4px`). But the parent site should really be calling the shots on styling outside the widget.

**Comment by joethreepwood** [comment]
Alright alright, border removed 😛 

When adding the FF in as above, I see three errors in VSC: 

On `inject`
```
Function implementation is missing or not immediately following the declaration.ts(2391)
function inject({ config, posthog }: {
    config: any;
    posthog: any;
}): any
```

Followed by `Property 'posthog' does not exist on type 'Window & typeof globalThis'.` twice on `if (window.posthog && window.posthog.getFeatureFlag('posthog-nps-survey-app') === true)`

There's no further error. When testing it locally, it just doesn't load the widget at all.

**Comment by joethreepwood** [comment]
Thinking on this, my assumption is that if the `if (window.posthog && window.posthog.getFeatureFlag('posthog-nps-survey-app') === true)` is set to `false` then the widget should still load _even_ if I haven't made the FF yet. However, this may not be true and it's actually still looking for a flag to be in the negative state, rather than the absence of a flag?

**Comment by ajsharp** [comment]
> Thinking on this, my assumption is that if the `if (window.posthog && window.posthog.getFeatureFlag('posthog-nps-survey-app') === true)` is set to `false` then the widget should still load _even_ if I haven't made the FF yet. However, this may not be true and it's actually still looking for a flag to be in the negative state, rather than the absence of a flag?

I was trying to figure out how to program _when_ to display this, and stumbled on this ticket, so I'll weigh in 🙃 

Personally, how I'd like to use this is to be able to customize when this displays via some function call from my own code, and I would handle the feature flag stuff from client-side code. So I'd rather handle that logic, personally.

Thanks for making this!
